frontend/middleware_validatestatic.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-logr/logr"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+const (
+	wantedResourceProviderNamespace = "Microsoft.RedHatOpenShift"
+	resourceTypeHCPOpenshiftCluster = "HcpOpenShiftClusters"
+)
+
+var rxResourceGroupName = regexp.MustCompile(`^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
+
+func MiddlewareValidateStatic(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+
+	subId := chi.URLParam(r, "subscriptionId")
+	resourceGroupName := chi.URLParam(r, "resourceGroupName")
+	resourceProviderNamespace := chi.URLParam(r, "resourceProviderNamespace")
+	resourceType := chi.URLParam(r, "resourceType")
+	operationId := chi.URLParam(r, "operationId")
+	resourceName := chi.URLParam(r, "resourceName")
+
+	apiVersion := r.URL.Query().Get(APIVersionKey)
+
+	if r.URL.Path != strings.ToLower(r.URL.Path) {
+		if log, ok := r.Context().Value(contextKeyOriginalPath).(logr.Logger); ok {
+			log.Error(errors.New("LowerCase error"), "path was not lower case")
+		}
+		arm.WriteError(w, http.StatusInternalServerError, arm.CloudErrorCodeInternalServerError, "", "Internal server error.")
+		return
+	}
+
+	if subId != "" {
+		valid := api.IsValid(subId)
+		if !valid {
+			arm.WriteError(w, http.StatusBadRequest, arm.CloudErrorCodeInvalidSubscriptionID, "", "The provided subscription identifier '%s' is malformed or invalid.", subId)
+			return
+		}
+	}
+
+	if resourceGroupName != "" {
+		if !rxResourceGroupName.MatchString(resourceGroupName) {
+			arm.WriteError(w, http.StatusBadRequest, arm.CloudErrorCodeResourceGroupNotFound, "", "Resource group '%s' could not be found.", resourceGroupName)
```

Comment:
this error message is misleading, we did not attempt a resourceGroup lookup to determine if it could be found. Can we rephrase this to align with the validation?